src/core/utils/EventNormalizer.ts

```diff
@@ -294,30 +296,73 @@ export class EventNormalizer {
                     : compiledEvent.key;
             if (data === ' ' || data === 'Space') {
                 // Some send space as ' ' and some send 'Space'.
-                this._triggerEvent('insert', '\u00A0', elements); // nbsp
+                this._sendSignal(
+                    this._formatSignal(
+                        'insert',
+                        {
+                            value: '\u00A0',
+                        },
+                        elements,
+                    ),
+                ); // nbsp
             } else if (data && data[0] === '\u000A') {
                 // The enter key on some mobile keyboards do not trigger an
                 // actual keypress event but trigger an input event with the
                 // LINE FEED (LF) (u000A) unicode character instead.
                 // TODO: replace this <br/> by a contextualized 'enter' event
-                this._triggerEvent('insert', '<br/>', elements);
+                this._sendSignal(
+                    this._formatSignal(
+                        'insert',
+                        {
+                            value: '<br/>',
+                        },
+                        elements,
+                    ),
+                );
             } else {
-                this._triggerEvent('insert', data, elements);
+                this._sendSignal(
+                    this._formatSignal(
```

Comment:
If you don't name the dispatcher function `_sendSignal`, you can implement `_sendSignal` such that it formats then send the signal, thus saving you one function call everytime... ;-)